app/services/manifest_validator.rb

```diff
@@ -0,0 +1,28 @@
+class ManifestValidator
+  include JsonSchemaHelper
+
+  attr_reader :content, :errors
+
+  def initialize(content)
+    if content.is_a?(String)
+      @content = JSON.parse(content)
+    else
+      @content = content
+    end
+    register_schemas_by_uri
+  end
+
+  def run
+    schema_file_path = schema_file('post-manifest')
+    @errors = JSON::Validator.fully_validate(schema_file_path, content, errors_as_objects: true)
+    !@errors.any?
```

Comment:
All of that is to say, returning a special hash means your code would have to look like:

```ruby
response = validator.run
if response[:errors] == true
  puts "You had problems: #{response.messages.join("\n")}"
end
```

which seems like it is both verbose and prone to error (what if you typo the hash key and check for `:errorz` instead? you might never know you had any problems).